lib/bypass.ex

```diff
@@ -4,41 +4,67 @@ defmodule Bypass do
   import Bypass.Utils
   require Logger
 
+
   def open(opts \\ []) do
+    open(:ex_unit, opts)
+  end
+
+  def open(framework, opts) do
     case Supervisor.start_child(Bypass.Supervisor, [opts]) do
       {:ok, pid} ->
         port = Bypass.Instance.call(pid, :port)
         debug_log "Did open connection #{inspect pid} on port #{inspect port}"
-        ExUnit.Callbacks.on_exit({Bypass, pid}, fn ->
-          case Bypass.Instance.call(pid, :on_exit) do
-            :ok ->
-              :ok
-            :ok_call ->
-              :ok
-            {:error, :too_many_requests, {:any, :any}} ->
-              raise ExUnit.AssertionError, "Expected only one HTTP request for Bypass"
-            {:error, :too_many_requests, {method, path}} ->
-              raise ExUnit.AssertionError, "Expected only one HTTP request for Bypass at #{method} #{path}"
-            {:error, :unexpected_request, {:any, :any}} ->
-              raise ExUnit.AssertionError, "Bypass got an HTTP request but wasn't expecting one"
-            {:error, :unexpected_request, {method, path}} ->
-              raise ExUnit.AssertionError,
-                "Bypass got an HTTP request but wasn't expecting one at #{method} #{path}"
-            {:error, :not_called, {:any, :any}} ->
-              raise ExUnit.AssertionError, "No HTTP request arrived at Bypass"
-            {:error, :not_called, {method, path}} ->
-              raise ExUnit.AssertionError,
-                "No HTTP request arrived at Bypass at #{method} #{path}"
-            {:exit, {class, reason, stacktrace}} ->
-              :erlang.raise(class, reason, stacktrace)
-          end
-        end)
+        setup_expectations_verifications(framework, pid)
         %Bypass{pid: pid, port: port}
       other ->
         other
     end
   end
 
+  defp setup_expectations_verifications(:ex_unit, pid) do
+    ExUnit.Callbacks.on_exit({Bypass, pid}, fn ->
+      do_verify_expectations(pid, ExUnit.AssertionError)
+    end)
+  end
+
+  defp setup_expectations_verifications(_framework, _pid), do: nil
+
+
+  if Code.ensure_loaded?(ESpec) do
```

Comment:
> I assume this is to prevent an xref warning?
Yes, to avoid compilation warnings but also runtime exceptions when trying to `raise` with an unknown identifier, below.

> What about excluding it explicitly in the mix.exs?
I know how to exclude a file but I'm not sure how to exclude a function, if this is what you meant.
`Code.ensure_loaded?/1` should be the standard way to do it, though.